asm-lsp/parser.rs

```diff
@@ -778,6 +779,147 @@ pub fn populate_6502_instructions(html_conts: &str) -> Result<Vec<Instruction>>
+///
+/// # Errors
+///
+/// This function is highly specialized to parse a single file and will panic or return
+/// `Err` for most mal-formed inputs
+///
+/// # Panics
+///
+/// This function is highly specialized to parse a single file and will panic or return
+/// `Err` for most mal-formed/unexpected inputs
+// NOTE:
+// Raw JSON file pruned via the command:
+// ```
+// jq ".instructions | map({mnemonics: .mnemonics | map(del(.intrinsics)), body})" power-isa.json
+// ```
+pub fn populate_power_isa_instructions(json_conts: &str) -> Result<Vec<Instruction>> {
+    #[allow(non_camel_case_types, clippy::upper_case_acronyms)]
+    #[derive(Deserialize, Debug, Copy, Clone)]
+    enum PowerReleaseRepr {
+        P1,
+        P2,
+        PPC,
+        #[serde(rename = "v2.00")]
+        v200,
+        #[serde(rename = "v2.01")]
+        v201,
+        #[serde(rename = "v2.02")]
+        v202,
+        #[serde(rename = "v2.03")]
+        v203,
+        #[serde(rename = "v2.04")]
+        v204,
+        #[serde(rename = "v2.05")]
+        v205,
+        #[serde(rename = "v2.06")]
+        v206,
+        #[serde(rename = "v2.07")]
+        v207,
+        #[serde(rename = "v3.0")]
+        v30,
+        #[serde(rename = "v3.0B")]
+        v30B,
+        #[serde(rename = "v3.0C")]
+        v30C,
+        #[serde(rename = "v3.1")]
+        v31,
+        #[serde(rename = "v3.1B")]
+        v31B,
+    }
+
+    impl PowerReleaseRepr {
+        fn release_message(self) -> String {
+            String::from(match self {
+                Self::P1 => "P1 instruction introduced in POWER Architecture",
+                Self::P2 => "P2 Instruction introduced in POWER2 Architecture",
+                Self::PPC => "PPC Instruction introduced in PowerPC Architecture prior to v2.00",
+                Self::v200 => "v2.00 Instruction introduced in PowerPC Architecture Version 2.00",
+                Self::v201 => "v2.01 Instruction introduced in PowerPC Architecture Version 2.01",
+                Self::v202 => "v2.02 Instruction introduced in PowerPC Architecture Version 2.02",
+                Self::v203 => "v2.03 Instruction introduced in Power ISA Version 2.03",
+                Self::v204 => "v2.04 Instruction introduced in Power ISA Version 2.04",
+                Self::v205 => "v2.05 Instruction introduced in Power ISA Version 2.05",
+                Self::v206 => "v2.06 Instruction introduced in Power ISA Version 2.06",
+                Self::v207 => "v2.07 Instruction introduced in Power ISA Version 2.07",
+                Self::v30 => "v3.0 Instruction introduced in Power ISA Version 3.0",
+                Self::v30B => "v3.0B Instruction introduced in Power ISA Version 3.0B",
+                Self::v30C => "v3.0C Instruction introduced in Power ISA Version 3.0C",
+                Self::v31 => "v3.1 Instruction introduced in Power ISA Version 3.1",
+                Self::v31B => "v3.1B Instruction introduced in Power ISA Version 3.1B",
```

Comment:
I think it is simpler to shorten each string to "Introduced in ...". Also, the list switches between ISA and Architecture when they pretty much mean the same thing in this context. Either remove them (ex: "PowerPC Version 2.01") or stick to one consistently. I think I gave you these strings so consider this my error haha. I think these changes would remove a little clutter from the hover view.

Reply:
Sounds good to me!